Source/player.cpp

```diff
@@ -2055,16 +2055,22 @@ void Player::UpdatePreviewCelSprite(_cmd_id cmdId, Point point, uint16_t wParam1
 	}
 }
 
+void Player::setCharacterLevel(uint8_t level)
+{
+	this->_pLevel = clamp<uint8_t>(level, 1U, MaxCharacterLevel);
+	this->_pNextExper = GetNextExperienceThresholdForLevel(this->getCharacterLevel());

```

Comment:
it's pretty much only set for the use in `ValidatePlayer()` since that's called every frame. I was planning on expanding #6399 to allow specifying separate experience thresholds (and max levels, still need to make that dynamic...) for each class which would make the lookup slightly more expensive but it's really not significant work overall. I'll do this suggestion this evening :D